### PR TITLE
fix: fix most typos

### PR DESCRIPTION
**Pull request by dsawp**
## Issue Addressed

I use some tools to check typos. This pr will fix most typos in this repo.

### CODE REVIEW

**Comment by dsawp** [comment]
Hi maintainers, please review :)

File: beacon_node/beacon_chain/src/canonical_head.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@
 //! 3. `Mutex<()>`: Is used to prevent concurrent execution of `BeaconChain::recompute_head`.
 //!
 //! This module has to take great efforts to avoid causing a deadlock with these three methods. Any
-//! developers working in this module should tread carefully and seek a detailed review.
+//! developers working in this module should treat carefully and seek a detailed review.
```
Comment (michaelsproul) [inline beacon_node/beacon_chain/src/canonical_head.rs]:
```suggestion
//! developers working in this module should tread carefully and seek a detailed review.
```

False positive. "Tread" is the intended word here and makes more sense than "treat".

File: beacon_node/http_api/tests/fork_tests.rs [hunk 2]
```diff
@@ -293,7 +293,7 @@ async fn sync_committee_indices_across_fork() {
         400
     );
 
-    // Once the head is updated it should be useable for requests, including in the next sync
+    // Once the head is updated it should be usable for requests, including in the next sync
```
Comment (michaelsproul) [inline beacon_node/http_api/tests/fork_tests.rs]:
```suggestion
    // Once the head is updated it should be useable for requests, including in the next sync
```

Useable is fine in en_GB and en_AU

File: beacon_node/network/src/sync/range_sync/chain_collection.rs [hunk 3]
```diff
@@ -84,7 +84,7 @@ impl<T: BeaconChainTypes, C: BlockStorage> ChainCollection<T, C> {
                         RangeSyncState::Head(syncing_head_ids)
                     };
                 } else {
-                    // we removed a head chain, or an stoped finalized chain
+                    // we removed a head chain, or an stopped finalized chain
```
Comment (michaelsproul) [inline beacon_node/network/src/sync/range_sync/chain_collection.rs]:
```suggestion
                    // we removed a head chain, or a stopped finalized chain
```

File: boot_node/src/server.rs [hunk 4]
```diff
@@ -17,7 +17,7 @@ pub async fn run<E: EthSpec>(
     eth2_network_config: &Eth2NetworkConfig,
     log: slog::Logger,
 ) -> Result<(), String> {
-    // parse the CLI args into a useable config
+    // parse the CLI args into a usable config
```
Comment (michaelsproul) [inline boot_node/src/server.rs]:
```suggestion
    // parse the CLI args into a useable config
```

File: watch/src/suboptimal_attestations/updater.rs [hunk 5]
```diff
@@ -21,7 +21,7 @@ impl<E: EthSpec> UpdateHandler<E> {
     /// That is, `if highest_filled_slot % slots_per_epoch != 31`.
     ///
     /// In the event the most recent epoch has no suboptimal attestations, it will attempt to
-    /// resync that epoch. The odds of this occuring on mainnet are vanishingly small so it is not
+    /// resync that epoch. The odds of this Unavailable on mainnet are vanishingly small so it is not
```
Comment (michaelsproul) [inline watch/src/suboptimal_attestations/updater.rs]:
```suggestion
    /// resync that epoch. The odds of this occurring on mainnet are vanishingly small so it is not
```

**Comment by michaelsproul** [comment]
Need to discuss with the team whether we want to merge this, as we are trying to discourage airdrop farming:

https://github.com/sigp/lighthouse/blob/unstable/CONTRIBUTING.md#i-dont-think-i-have-anything-to-add

> **Please note, to maintain project quality, we may not accept PRs for small typos or changes with minimal impact.**

**Comment by michaelsproul** [comment]
Sorry but we have decided not to accept this PR or any other typo fix PRs. We will clarify the policy in the docs.

**Pull request closed without merging**